### ISSUE TITLE
Incorrect directory name in opentelemetry-dotnet-instrumentation-linux-musl release

### ISSUE THREAD

**Issue by rajkumar-rangaraj** [open-telemetry/opentelemetry-dotnet-instrumentation#2179]
All of our documentation recommends to use `/linux-musl-x64/OpenTelemetry.AutoInstrumentation.Native.so` for `CORECLR_PROFILER_PATH`. In our latest release, directory name for alpine version is listed as `linux-64`, this should break our `instrument.sh` scripts too.

![image](https://user-images.githubusercontent.com/9479006/217707269-8979b91e-ad35-45a8-889c-74d14c2bf3a1.png)

We might need to extend our `DistributionStructure` tests to run in release pipeline.

**Comment by Kielek** [comment]
@rajkumar-rangaraj, 
I have found the reason for this
we have dedicated, short-running, release step for this. Unfortunately, OS_TYPE is not configured there
see https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/3fd02e17a1aa88fde93a207d9c32c7cf95ba13ab/.github/workflows/ci.yml#L105-L111 and https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/3fd02e17a1aa88fde93a207d9c32c7cf95ba13ab/.github/workflows/ci.yml#L105-L111.

Proposal how to fix it
1. Temporary fix the release pipeline.
2. I suppose that we should avoid modifying release packages, so we should shortly release beta.2 with this fix.
3. Long-term use the same pipeline (with executing tests) as for PRs/main. See example here https://github.com/signalfx/splunk-otel-dotnet/blob/e5ceb285b6c1c69c5a3ea12efcf097dbf6dd3e78/.github/workflows/ci.yml#L105
